pkg/controller/metrics/kluster.go

```diff
@@ -65,6 +65,8 @@ func (collector *klusterCollector) Collect(ch chan<- prometheus.Metric) {
 			kluster.Status.ChartName,
 			getCreatorFromAnnotations(kluster.Annotations),
 			getAccountFromLabels(kluster.Labels),
+			kluster.Spec.Backup,
+			*kluster.Spec.Audit,
```

Comment:
Is this guaranteed to be non-nil all the time? Otherwise this would create a panic

Reply:
As per typing it cannot (if my novice understanding of go is correct):
`invalid operation: cannot compare kluster.Spec.Backup == nil (mismatched types string and untyped nil)`
Did I miss a check?

See `kluster_spec.go`:
https://github.com/sapcc/kubernikus/blob/master/pkg/api/models/kluster_spec.go#L30-L36

Reply:
`Backup` is fine because it not a pointer, but `Audit` is. dereferencing a pointer that is not set (`*pointername`) creates a panic crashing the process.

Reply:
I was wondering why `Audit` was a pointer and Backup not. The `kluster_spec.go` is autogenerated by swagger right? So I guess this is the way of marking optional properties?

Reply:
Setting audit is optional, so null is valid value, which means do not audit log.https://github.com/sapcc/kubernikus/blob/master/swagger.yml#L472